modules/ding_availability/ding_availability.module

```diff
@@ -91,22 +91,39 @@ function ding_availability_theme($existing, $type, $theme, $path) {
 
 /**
  * Output the availability information as json.
- *
- * @param array $provider_ids
- *   Providers ID's to fetch availability information for.
  */
-function ding_availability_js($provider_ids) {
-  drupal_json_output(ding_availability_items(explode(',', $provider_ids)));
+function ding_availability_js() {
+  $data = isset($_POST['ids']) ? $_POST['ids'] : array();
+
+  // Enrich the return data with ids to simplify the work need in the frontend
+  // to insert the holdings.
+  $items = ding_availability_items($data);
+  foreach ($items as $provider_id => &$item) {
+    foreach ($data as $ids) {
+      if ($ids['local'] == $provider_id) {
+        $item['ids'] = $ids;
+        break;
+      }
+    }
+  }
+
+  drupal_json_output($items);
 }
 
 /**
  * Output holdings information as json.
- *
- * @param array $provider_ids
- *   Providers ID's to fetch holdings information for.
  */
-function ding_availability_holdings_js($provider_ids) {
-  drupal_json_output(ding_availability_holdings(explode(',', $provider_ids)));
+function ding_availability_holdings_js() {
+  $data = isset($_POST['ids']) ? $_POST['ids'] : array();
+
+  // Enrich the return data with ids to simplify the work need in the frontend
+  // to insert the holdings.
+  $holdings = ding_availability_holdings($data);
+  foreach ($holdings as $provider_id => &$holding) {
+    $holding['ids'] = $data[$provider_id];
+  }
```

Comment:
This might be possible to change to do it in the JS code. But there is so much legacy in the JS part and to be honest I ran out-of-time to fix it all.

If you think that this should be fixed I'll ask DDB for more time.